Source/WebCore/inspector/InspectorInstrumentation.cpp

```diff
@@ -79,14 +80,26 @@
 #include <JavaScriptCore/InspectorDebuggerAgent.h>
 #include <JavaScriptCore/ScriptArguments.h>
 #include <JavaScriptCore/ScriptCallStack.h>
+#include <memory>
+#include <wtf/NeverDestroyed.h>
+#include <wtf/Observer.h>
 #include <wtf/StdLibExtras.h>
+#include <wtf/Vector.h>
+#include <wtf/WeakHashSet.h>
 
 namespace WebCore {
 
 using namespace Inspector;
 
 namespace {
 static HashSet<InstrumentingAgents*>* s_instrumentingAgentsSet = nullptr;
+#if ENABLE(WEBDRIVER_BIDI)
+static WeakHashSet<InspectorInstrumentationConsoleMessageObserver>& consoleMessageObservers()
```

Comment:
this is likely going to be used in for a lot more things than just console messages.  furthermore, i'm not sure i like the blending of Web Inspector and WebDriver things.  id much prefer if we created a new `AutomationInstrumentation` or something like that and used it in the same places that there's `InspectorInstrumentation` instead.  this way, we can adjust each independently of each other as needed.

Reply:
Right. I added a new `AutomationInstrumentation`, with just the new `addMessageToConsole`.